### PR TITLE
Plt1670 - Add only package name and version as label and annotation

### PR DESCRIPTION
**Pull request by anbarasantr**
- Package name and version contains special character and it cannot be added in K8s pod labels. so its added as annotations

Screenshot 2024-05-30 at 3 17 09 PM

Screenshot 2024-05-30 at 3 16 47 PM

### CODE REVIEW

File: lib/k8s.js [hunk 1]
```diff
@@ -78,13 +78,22 @@ function enableClusterScope(yamlObject) {
...
+
     if (kind === constants.ARGO_WORKFLOW_TEMPLATES_KIND || kind === constants.ARGO_CLUSTER_WORKFLOW_TEMPLATES_KIND) {
         let metadata = yamlObject.metadata;
         let spec = yamlObject.spec;
-        // Copy labels from metadata to spec.podMetadata.labels
         spec.podMetadata = spec.podMetadata || {};
-        spec.podMetadata.labels = Object.assign({}, spec.podMetadata.labels, metadata.labels);
+        if (metadata.labels && metadata.labels[ANNOTATION_VERSION]) {
+            spec.podMetadata.labels = spec.podMetadata.labels || {};
+            spec.podMetadata.labels[ANNOTATION_VERSION] = metadata.labels[ANNOTATION_VERSION];
```
Comment (louisnow) [inline lib/k8s.js:91]: Canary version might have special characters in version

Comment (anbarasantr) [reply lib/k8s.js:91]: Good catch.

File: lib/k8s.js [hunk 2]
```diff
@@ -78,13 +78,22 @@ function enableClusterScope(yamlObject) {
...
         let spec = yamlObject.spec;
-        // Copy labels from metadata to spec.podMetadata.labels
         spec.podMetadata = spec.podMetadata || {};
-        spec.podMetadata.labels = Object.assign({}, spec.podMetadata.labels, metadata.labels);
+        if (metadata.labels && metadata.labels[ANNOTATION_VERSION]) {
+            spec.podMetadata.annotations = spec.podMetadata.annotations || {};
+            spec.podMetadata.annotations[ANNOTATION_VERSION] = metadata.labels[ANNOTATION_VERSION];
+        }
+        if (metadata.annotations && metadata.annotations[LABEL_NAME]) {
+            spec.podMetadata.annotations = spec.podMetadata.annotations || {};
```
Comment (louisnow) [inline lib/k8s.js:94]: This line resets the work done in the previous if block right?

Comment (anbarasantr) [reply lib/k8s.js:94]: Not really. 
spec.podMetadata.annotations is initialized to an empty object only if it is undefined or null.
I have a tests that clarifies this

**Review by louisnow — APPROVED** [review]

**Pull request merged**